lib/fastlane/actions/update_app_identifier.rb

```diff
@@ -0,0 +1,89 @@
+          project_path = params[:xcodeproj]
+          project = Xcodeproj::Project.open(project_path)
+
+          # Fetch the build configuration objects
+          configs = project.objects.select { |obj| obj.isa == 'XCBuildConfiguration' && !obj.build_settings[identifier_key].nil? }
+          raise "Info plist uses $(#{identifier_key}), but xcodeproj does not".red unless configs.count > 0
+
+          # For each of the build configurations, set app identifier
+          configs.each do |c|
+            c.build_settings[identifier_key] = params[:app_identifier]
+          end
+
+          # Write changes to the file
+          project.save
+
+          Helper.log.info "Updated #{params[:xcodeproj]} 💾.".green
+        else
+          # Update plist value
+          plist['CFBundleIdentifier'] = params[:app_identifier]
+
+          # Write changes to file
+          plist_string = Plist::Emit.dump(plist)
+          File.write(info_plist_path, plist_string)
+
+          Helper.log.info "Updated #{params[:plist_path]} 💾.".green
+        end
+      end
+
+      #####################################################
+      # @!group Documentation
+      #####################################################
+
+      def self.is_supported?(platform)
+        [:ios].include?(platform)
+      end
+
+      def self.description
+        'Update an app identifier'
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(key: :xcodeproj,
+                                       env_name: "FL_UPDATE_APP_IDENTIFIER_PROJECT_PATH",
+                                       description: "Path to your Xcode project",
+                                       optional: true,
+                                       verify_block: proc do |value|
+                                         raise "Please pass the path to the project, not the workspace".red if value.include? "workspace"
+                                         raise "Could not find Xcode project".red unless File.exist?(value)
+                                       end),
+          FastlaneCore::ConfigItem.new(key: :plist_path,
+                                       env_name: "FL_UPDATE_APP_IDENTIFIER_PLIST_PATH",
+                                       description: "Path to info plist, relative to your Xcode project",
+                                       verify_block: proc do |value|
+                                         raise "Invalid plist file".red unless value[-6..-1].downcase == ".plist"
+                                       end),
+          FastlaneCore::ConfigItem.new(key: :app_identifier,
+                                       env_name: 'FL_UPDATE_APP_IDENTIFIER',
+                                       description: 'The App Identifier of your app',
+                                       default_value: ENV['PRODUCE_APP_IDENTIFIER'])
+        ]
+      end
+
+      def self.authors
+        ['squarefrog', 'tobiasstrebitzer']
```

Comment:
Indirectly - I wrote that so long ago I honestly can't remember why. IIRC it was because part of an existing `fastlane` action helped with the code of this action.